### PR TITLE
Feature/mptcpv1 - MPTCP scripts and integration into mesh schield

### PR DESCRIPTION
**Pull request by Monika-Research**
https://github.com/tiiuae/tc_distro/pull/139 - dependent PR

Initd script for MPTCP
Starting S90mptcp process if required
MPTCP script for configuring routing rules and starting the proxy server

### CODE REVIEW

**Review by joenpera — COMMENTED** [review]
what if new configuration is sent to device and new configuration is applied and processes restarted with __apply_mission_config() (comms_command.py)?

/var/run/mptcp.conf changes, but is the S90mptcp aware of those changes?

**Review by saauvine — APPROVED** [review]

**Comment by Monika-Research** [comment]
> what if new configuration is sent to device and new configuration is applied and processes restarted with __apply_mission_config() (comms_command.py)?
> 
> /var/run/mptcp.conf changes, but is the S90mptcp aware of those changes?

Following commits made to handle new mission configs
https://github.com/tiiuae/mesh_com/pull/333/commits/8a54c46f715d482cfe44627080f6214284b94903
https://github.com/tiiuae/mesh_com/pull/333/commits/b95019c8540c032e740d6a7de7696ef3d0b2c7ab
https://github.com/tiiuae/mesh_com/pull/333/commits/3142bda50d840825d24aa8bd781c939382ed341f
https://github.com/tiiuae/mesh_com/pull/333/commits/b168f2fee3618ff14f732e002d427739f0d48b3b

File: modules/sc-mesh-secure-deployment/src/nats/src/comms_command.py [hunk 1]
```diff
@@ -245,6 +245,15 @@ def __apply_mission_config(self) -> Tuple[str, str]:
...
             ]
+            # Stop S90mptcp before restarting S9011sNatsMesh
+            ret = subprocess.run(
+                ["/opt/S90mptcp", "stop"], shell=False, check=True, capture_output=True,)
+            if ret.returncode != 0:
+                return "FAIL", f"Clearing MPTCP configs failed" + str(
+                    ret.returncode
+                ) + str(ret.stdout) + str(ret.stderr)
+            self.logger.debug("Stopped MPTCP service")
+            return "OK", "Stopped MPTCP service"
```
Comment (joenpera) [inline modules/sc-mesh-secure-deployment/src/nats/src/comms_command.py:256]: MPTCP stopped and returned without restarting mesh and starting mptcp?

File: modules/sc-mesh-secure-deployment/src/nats/src/comms_command.py [hunk 2]
```diff
@@ -260,6 +269,15 @@ def __apply_mission_config(self) -> Tuple[str, str]:
...
             return "OK", "Mission configurations applied"
+            # Start S90mptcp after restarting S9011sNatsMesh
+            ret = subprocess.run(
+                ["/opt/S90mptcp", "start"], shell=False, check=True, capture_output=True, )
+            if ret.returncode != 0:
+                return "FAIL", f"Setting MPTCP configs failed" + str(
+                    ret.returncode
+                ) + str(ret.stdout) + str(ret.stderr)
+            self.logger.debug("Started MPTCP service")
+            return "OK", "Started MPTCP service"
```
Comment (joenpera) [inline modules/sc-mesh-secure-deployment/src/nats/src/comms_command.py:280]: This code is never executed as there is return "OK" above already

Comment (joenpera) [reply modules/sc-mesh-secure-deployment/src/nats/src/comms_command.py:281]: Apply command should return --> return "OK", "Mission configurations applied"

Comment (Monika-Research) [reply modules/sc-mesh-secure-deployment/src/nats/src/comms_command.py]: Changed the return message and also added S90mptcp stop when revoke applied.
https://github.com/tiiuae/mesh_com/commit/401159126efb1a826aaf73f50ac630186877c3ca

**Review by joenpera — APPROVED** [review]

**Review by mbaddeley — APPROVED** [review]

**Pull request merged**